Refactor of Canvas classes

jonatanschroeder opened the pull request.

steven-wolfman committed:
Added a utility method to update an assignment

steven-wolfman committed:
added token.txt (proposed as default token location)

steven-wolfman committed:
new script to process text in Canvas (e.g., assignment descriptions or page contents)

steven-wolfman committed:
abstracting common behavior from quizzes, assignments, and pages

steven-wolfman committed:
page fetching for Course

steven-wolfman committed:
typo bugfix

steven-wolfman committed:
patching fetch of pages to grab pagebodies explicitly

steven-wolfman committed:
many updates: minimally functioning prototype now

steven-wolfman committed:
Refactored ObjectType class into multiple higher-order functions, prepping for handling quiz questions

steven-wolfman committed:
unix style line endings (I hope!)

steven-wolfman committed:
pep8 pylint run (mostly line endings and blank lines)

steven-wolfman committed:
incomplete quiz question commit

steven-wolfman committed:
bug fix: get_course

steven-wolfman committed:
bugfix: do not rename canvas import!

steven-wolfman committed:
closer to handling quiz questions; still fails with an error

steven-wolfman committed:
debugged various errors (super vs super(), missing self parameter, etc.)

steven-wolfman committed:
spacing in logs

steven-wolfman committed:
Merge remote-tracking branch 'upstream/master'

steven-wolfman committed:
patch merge error fast-forwarding to Jonatan's current version (f-strings vs. old style Python formatting strings)

jonatanschroeder commented on canvas.py:
```diff
@@ -364,4 +463,15 @@ def update_rubric(self, rubric):
+                         id_field="url", request_param_name="wiki_page")
+
+    def update_page(self, data=None):
+        return self.update(self, data)
```
```suggestion
        return self.update(data)
```

steven-wolfman replied:
Oops! Indeed; I'll commit that.

jonatanschroeder reviewed (commented):
Looking good... @steven-wolfman can you just check the code fix I listed in the comments?

steven-wolfman committed:
Eliminate mistaken explicit self parameter

Thanks to Jonatan for pointing this out! I did a quick check for similar mistakes (via regex search for "self[,)]") and didn't find anything.

Co-authored-by: Jonatan Schroeder <jonatan@cs.ubc.ca>
